automata/pda/pda.py

```diff
@@ -258,6 +262,58 @@ def show_diagram(
 
         return graph
 
+    def _create_stack_diagram(
+        self,
+        input_path: List[Tuple[PDAConfiguration, PDAConfiguration]],
+        graph: pgv.AGraph,
+        start_color: coloraide.Color,
+        end_color: coloraide.Color,
+        font_size: str,
+        arrow_size: str,
+    ) -> pgv.AGraph:
+        """
+        Constructs stack for all the transitions in the `input_path` and
+        adds the constructed stacks into `graph`. Returns the same `graph`
+        """
+        from_node = input_path[0][0]
+        label = " | ".join(reversed(from_node.stack))  # type: ignore[arg-type]
```

Comment:
`mypy` gives an error here, so I have put an ignore comment. If you want me to remove the ignore comment that can be done. I will change the code to `reversed(from_node.stack.stack)` that will resolve the error from `mypy`.

The `mypy` error is:
```
automata/pda/pda.py:279: error: Argument 1 to "reversed" has incompatible type "PDAStack"; expected "SupportsLenAndGetItem[str]"  [arg-type]
automata/pda/pda.py:279: note: Following member(s) of "PDAStack" have conflicts:
automata/pda/pda.py:279: note:     Expected:
automata/pda/pda.py:279: note:         def __getitem__(self, int, /) -> str
automata/pda/pda.py:279: note:     Got:
automata/pda/pda.py:279: note:         def __getitem__(self, Union[int, slice], /) -> Union[str, Sequence[str]]
```

Reply:
@Vipul-Cariappa Per the docs for `reversed()`, you should be able to eliminate the need for the ignore comment by adding a `__reversed__` method to the `PDAStack` class.

```python
def __reversed__(self) -> Iterator[str]:
    return reversed(self.stack)
```

This method should've really been present anyway, so I would ask that you please go ahead and add it to `PDAStack`.